frb_codegen/src/generator/dart/func.rs

```diff
@@ -126,6 +127,11 @@ pub(crate) fn generate_api_func(
     } else {
         format!("_wire2api_{}", func.output.safe_ident())
     };
+    if ir_file.shared {
+        parse_success_data = parse_success_data.replace("_wire2api_", "wire2api_");
+    } else if ir_file.is_type_shared_by_safe_ident(&func.output) {
+        parse_success_data = parse_success_data.replace("_wire2api_", "_sharedImpl.wire2api_");
+    }
```

Comment:
it would be great if we deal with structured data, instead of raw strings

Reply:
I would suggest not doing any refactoring at present, since it is not clear what kind of struct should be used. And whether the so-called refactored struct would violate further code generation in future.